clippy_utils/src/lib.rs

```diff
@@ -861,14 +861,16 @@ pub fn get_enclosing_block<'tcx>(cx: &LateContext<'tcx>, hir_id: HirId) -> Optio
     })
 }
 
-/// Gets the loop enclosing the given expression, if any.
-pub fn get_enclosing_loop(tcx: TyCtxt<'tcx>, expr: &Expr<'_>) -> Option<&'tcx Expr<'tcx>> {
+/// Gets the loop or closure enclosing the given expression, if any.
+pub fn get_enclosing_loop_or_closure(tcx: TyCtxt<'tcx>, expr: &Expr<'_>) -> Option<&'tcx Expr<'tcx>> {
```

Comment:
It might be better to split this up into two functions to make it more versatile. Something like: `get_enclosing_loop` and `get_enclosing_closure`

The rest of the code and logic LGTM :upside_down_face:

Reply:
Only the innermost of the two is needed for the lint, so it can't really be two separate functions.

Reply:
Okay, that makes sense. In that case you can resolve this comment :)